### PR TITLE
Move openstack terraform configs into modules

### PR DESCRIPTION
**Pull request by stevendborrelli**
Based on supporting production clusters, we've found there are too many cases where change in the terraform configurations would destroy the cluster before re-provisioning. Existing openstack tf configurations will be marked for deprecation in a future version of mantl. 

By moving to modules we can: 
- reduce the number of destroy/rebuilds in terraform
- attach instances to existing networks and routers
- have floating ips only for the edge nodes
- roll out new groups of servers without affecting existing instances. This allows us to migrate to new versions of mantl without affecting production.
- dynamically create new roles for ansible/terraform.py using `role=`
- adds support for optional `user_data` to instance groups for cloud_init

This PR also clears out openstack vars in the roles, allowing us to use environment variables, addressing #864.

### CODE REVIEW

File: terraform/openstack-modules.sample.tf [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+variable name { default = "mantl" }     # resources will start with "mantl-"
+variable control_count { default = "3"} # mesos masters, zk leaders, consul servers
+variable worker_count { default = "5"}  # worker nodes
+variable edge_count { default = "2"}    # load balancer nodes
+
+# Run 'nova network-list' to get these names and values
+# Floating ips are optional
+variable external_network_uuid { default = "uuid-of-your-external-network" }
+variable floating_ip_pool { default = "name-of-your-floating-ip-pool" }
```
Comment (BrianHicks) [inline terraform/openstack-modules.sample.tf]: would it make sense to move these two defaults to a comment? That way, Terraform would complain if they weren't set. As it is, i'll try to use `uuid-of-your-external-network` as the content of the variable if not set by the user.

**Comment by ohal** [comment]
please test this

**Comment by siddharthist** [comment]
I love the inline comments! I was able to correctly configure the sample terraform file in just one go. I'm testing now, is there anything in particular you'd like me to check out (e.g. how might one add nodes to an existing network, etc.)?

**Comment by stevendborrelli** [comment]
@siddharthist yes, you could test using an existing network_uuid, or having another stanza for a separate group of worker nodes.

**Comment by siddharthist** [comment]
I was able to deploy this branch as I would any other on Openstack after configuring the sample file, and I was able to add another group of workers after rolling out the first deployment. I'm now rebasing this onto master, removing the old sample terraform config, and adding depreciation warnings to the old modules.

**Pull request merged**